doc/RichTextArea/RichTextArea.md

```diff
@@ -0,0 +1,742 @@
+```
+
+Creating a read-only informational control should also be easy:
+
+```java
+        SimpleViewOnlyStyledModel m = new SimpleViewOnlyStyledModel();
+        // add text segment using CSS style name (requires a style sheet)
+        m.addSegment("RichTextArea ", null, "HEADER");
+        // add text segment using direct style
+        m.addSegment("Demo", "-fx-font-size:200%;", null);
+        // newline
+        m.nl();
+
+        RichTextArea t = new RichTextArea(m);
+```
+
+
+## Description
+
+Two new controls are provided: **RichTextArea** and **CodeArea**.  RichTextArea works with styled text and embedded Nodes, whereas CodeArea is limited to plain text documents using single font, which allows for syntax highlighting.
+
+The data model (document) is separated from the control, allowing for greater flexibility.  **EditableRichTextModel** is a default model for RichTextArea, **CodeTextModel** is a default model for CodeArea.
+
+The following diagram illustrates the logical relationships between important classes:
+
+```
+RichTextArea                   control
+ ├─ TextPos                    immutable text position
+ ├─ SelectionSegment           single selection segment (two markers)
+ │   └─ Marker                 moveable text position within the document
+ │
+ ├─ StyledTextModel            document data model
+ │   ├─ RichParagraph          presents the paragraph contents to view
+ │   │   ├─ StyleAttrs         immutable map of style attributes 
+ │   │   ├─ StyledSegment      immutable styled segment
+ │   └─ DataFormatHandler      converter for import/export/clipboard operations
+ │       ├─ StyledInput        input stream of StyledSegments
+ │       ├─ StyledOutput       output stream of StyledSegments
+ │       └─ StyleResolver      converts external CSS styles to attributes
+ │
+ └─ RichTextAreaSkin           default skin
+     └─SideDecorator           right/left paragraph decorator factory
+```
+
+CodeArea extends RichTextArea and brings a few additional classes into the picture:
+
+```
+CodeArea                       control, extends RichTextArea
+ ├─ CodeTextModel              document model, extends BasePlainTextModel, extends StyledTextModel
+ └─ SyntaxDecorator            interface which provides styling for underlying plain text
+```
+
+The code is currently being incubated in the **javafx.incubator.controls module**.
+While this document makes an attempt to give an overview of various parts, please refer to the
+[API Specification](javadoc/javadoc.zip) for more detail.
+
+
+
+
+### Design Principles
+
+- paragraph-oriented model, up to ~2 billion rows
+- virtualized text cell flow
+- supports text styling with an application stylesheet or inline attributes
+- supports multiple views connected to the same model
+- single selection
+- input map allows for easy behavior extension
+
+
+
+### RichTextArea Properties
+
+The new **RichTextArea** control exposes the following properties:
```

Comment:
Is there any particular reason that we do not have lineSpacing property in RichTextArea and only in CodeArea? Do you think it is needed in RichTextArea as well?

Reply:
yes: the line spacing is a paragraph attribute in the rich text area, similarly to any other rich text editor.

In the CodeArea, we are dealing with one font, so it makes more sense to have it as a control-level property.